runtime/src/main/java/io/quarkus/jsch/runtime/JSchProxyConfig.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkus.jsch.runtime;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigDocDefault;
+import io.quarkus.runtime.annotations.ConfigGroup;
+
+@ConfigGroup
+public interface JSchProxyConfig {
+
+    public static final Boolean DEFAULT_ENABLED = true;
+    public static final Integer DEFAULT_PORT = 80;
+
+    /**
+     * Enable the proxy for the JSch session.
+     *
+     * <p>
+     * Defaults to {@code true} if host is set.
+     */
+    Optional<Boolean> enabled();
+
+    /**
+     * The host to use for the JSch proxy.
+     */
+    String host();
+
+    /**
+     * The port to use for the JSch proxy.
+     */
+    @ConfigDocDefault("80")
+    Optional<Integer> port();
```

Comment:
```suggestion
    OptionalInt port();
```

Reply:
We cannot use `@WithDefault` if the parent property is `Optional`
And `proxy.auth` is an optional value

cf:

* https://github.com/quarkusio/quarkus/issues/36150
* https://github.com/quarkusio/quarkus/issues/7862

Reply:
Ok, then better use `OptionalInt` instead